File: 03_09_2024_Sum_of_Digits_of_String_After_Convert.java
class Solution {
    public int getLucky(String s, int k) {
        if (s.equals("")) {
            return 0;
        }

        StringBuilder num = new StringBuilder();
        for (char c : s.toCharArray()) {
            int n = c - 'a' + 1;
            num.append(n);
        }
        
        int ans = 0;
        while (k > 0) {
            ans = 0;
            for (char digit : num.toString().toCharArray()) {
                ans += digit - '0';
            }
            num = new StringBuilder(String.valueOf(ans));
            k--;
        }
        return ans;
    }
}

File: 03_09_2024_Sum_of_Digits_of_String_After_Convert.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        if(s=="")
        return 0;

        string num = "";
        for(char c : s)
        {
            int n = c-'a'+1;
            num += to_string(n);
        }
        int ans = 0;
        while(k>0)
        {
            ans = 0;
            for (char digit : num) {
                ans += digit - '0';
            }
            num = to_string(ans);
            k--;
        }
        return ans;
    }
};
